crates/runner-host/crates/web-client/src/board_components/button.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use log::info;
+use web_common::{ButtonState, Event};
+use yew::prelude::*;
+
+#[derive(Properties, PartialEq)]
+pub struct Props {
+    #[prop_or_default]
+    pub id: usize,
+    pub on_event: Callback<Event>,
+}
+
+#[function_component]
+pub fn Button(Props { id, on_event }: &Props) -> Html {
+    let pressed = use_state(|| false);
```

Comment:
nit: usize is copy, so we can dereference it only once.
```suggestion
    let id = *id;
    let pressed = use_state(|| false);
```